presto-main/src/main/java/com/facebook/presto/server/WebUiResource.java

```diff
@@ -14,20 +14,31 @@
 package com.facebook.presto.server;
 
 import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
 import javax.ws.rs.Path;
+import javax.ws.rs.core.Context;
 import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
 
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.net.HttpHeaders.X_FORWARDED_PROTO;
 import static javax.ws.rs.core.Response.Status.MOVED_PERMANENTLY;
-import static javax.ws.rs.core.UriBuilder.fromPath;
 
 @Path("/")
 public class WebUiResource
 {
     @GET
-    public Response redirectIndexHtml()
+    public Response redirectIndexHtml(
+            @HeaderParam(X_FORWARDED_PROTO) String proto,
+            @Context UriInfo uriInfo)
     {
+        if (isNullOrEmpty(proto)) {
+            proto = uriInfo.getRequestUri().getScheme();
+        }
+
         return Response.status(MOVED_PERMANENTLY)
-                .location(fromPath("/ui/").build())
+                .location(UriBuilder.fromPath("/ui/").scheme(proto).build())
```

Comment:
static import `fromPath` (as it was)

Reply:
It's not because your change qualifies the call with `UriBuilder.`. Please add a static import and remove `UriBuilder.` from the call site. Please compare how this looked like before your change.

Reply:
@findepi  do you want me to push this change as part of another commit with same git message or should I squash two commits as part of single commit?

Reply:
please squash into a single commit. It's OK to push-force in case of this PR.